Input/GamepadTest/source/Gamepad.hx

```diff
@@ -2,195 +2,274 @@ package;
+			label.x += offset.x;
+			label.y += offset.y;
+			offset.putWeak();
+		}
+		inputLabels.add(label);
+		
+		switch(type)
+		{
+			case Digital: sprite.alpha = ALPHA_OFF;
+			case Invisible: sprite.visible = false;
+			default:
+		}
+		
+		inputSprites.push({ sprite:sprite, input:input, type:type, label:label });
+		return sprite;
 	}
 
-	function createBar(x:Float, y:Float):FlxBar
+	function createBar(x:Float, y:Float, label:String):Bar
 	{
 		var bar = new FlxBar(x, y, FlxBarFillDirection.LEFT_TO_RIGHT, 100, 14, null, "", 0, 100, true);
 		bar.createGradientBar([FlxColor.BLACK], [FlxColor.RED, FlxColor.BLUE], 1, 180, true, FlxColor.BLACK);
 		add(bar);
-		return bar;
+		var label = createLabel(x, y, label);
+		add(label);
+		return { bar:bar, label:label };
 	}
 
-	function createLabel(x:Float, y:Float, ?label:String):FlxText
+	function createLabel(x:Float, y:Float, label:String, size:Int = 8):FlxText
 	{
-		var label = new FlxText(x, y, 0, label);
+		var label = new FlxText(x, y, 0, label, size);
 		label.color = FlxColor.WHITE;
-		label.setBorderStyle(FlxTextBorderStyle.OUTLINE_FAST, FlxColor.BLACK, 1, 1);
-		add(label);
+		label.setBorderStyle(FlxTextBorderStyle.OUTLINE, FlxColor.BLACK, size/8, size/8);
+		label.autoSize = true;
 		return label;
 	}
-
+	
+	function drawLine(x:Float, y:Float, offset:FlxVector, color:FlxColor):FlxSprite
+	{
+		var line = new FlxSprite(x, y);
+		line.makeGraphic(Math.floor(offset.length), 2, color);
+		line.angle = offset.degrees;
+		line.origin.set(1, 1);
+		return line;
+	}
+	
+	inline function createStick(x:Float, y:Float, input:InputID)
+	{
+		var isLeft = input == LEFT_ANALOG_STICK;
+		var stick = createInputSprite(x, y, "Stick", input, Stick);
+		x = stick.x - this.x;// prevent double nested offset from spritegroup
+		y = stick.y - this.y;
+		var click = isLeft ? LEFT_STICK_CLICK : RIGHT_STICK_CLICK;
+		createInputSprite(x - 10, y - 10, "StickClick", click, Invisible, FlxVector.get(isLeft ? -70 : 80, 20));
+		if (isLeft)
+		{
+			createDirectionArrows(x, y, 40,
+				LEFT_STICK_DIGITAL_UP, LEFT_STICK_DIGITAL_DOWN, LEFT_STICK_DIGITAL_LEFT, LEFT_STICK_DIGITAL_RIGHT);
+		}
+		else
+		{
+			createDirectionArrows(x, y, 40,
+				RIGHT_STICK_DIGITAL_UP, RIGHT_STICK_DIGITAL_DOWN, RIGHT_STICK_DIGITAL_LEFT, RIGHT_STICK_DIGITAL_RIGHT);
+		}
+	}
+	
+	inline function createDirectionArrows(x:Float, y:Float, radius:Float, up:InputID, down:InputID, left:InputID, right:InputID)
+	{
+		createArrow(x, y - radius, up, -90);
+		createArrow(x, y + radius, down, 90);
+		createArrow(x - radius, y, left, 180);
+		createArrow(x + radius, y, right, 0);
+	}
+	inline function createArrow(x:Float, y:Float, input:InputID, angle:Float)
+	{
+		createInputSprite(x - 10, y - 10, "Arrow", input, Invisible).angle = angle;
+	}
+	
+	public function setActiveGamepad(gamepad:FlxGamepad):Void
+	{
+		this.gamepad = gamepad;
+		updateLabels();
+	}
+	
+	public inline function updateLabels():Void
```

Comment:
This doesn't seem like a good use of `inline` either.

Reply:
this one looks like a leftover from a previous implementation